lib/matplotlib/colors.py

```diff
@@ -1226,6 +1255,866 @@ def reversed(self, name=None):
+        cmapobject._rgba_bad = np.copy(self._rgba_bad)
+        cmapobject._shape = self.shape
+        if self._isinit:
+            cmapobject._lut = np.copy(self._lut)
+        return cmapobject
+
+    def __eq__(self, other):
+        if not isinstance(other, BivarColormap):
+            return False
+        # To compare lookup tables the Colormaps have to be initialized
+        if not self._isinit:
+            self._init()
+        if not other._isinit:
+            other._init()
+        if not np.array_equal(self._lut, other._lut):
+            return False
+        if not np.array_equal(self._rgba_bad, other._rgba_bad):
+            return False
+        if not np.array_equal(self._rgba_outside, other._rgba_outside):
+            return False
+        if self.shape != other.shape:
+            return False
+        return True
+
+    def get_bad(self):
+        """Get the color for masked values."""
+        return self._rgba_bad
+
+    def get_outside(self):
+        """Get the color for out-of-range values."""
+        return self._rgba_outside
+
+    def resampled(self, lutshape, transposed=False):
+        """
+        Return a new colormap with *lutshape* entries.
+
+        Note that this function does not move the origin.
+
+        Parameters
+        ----------
+        lutshape : tuple of ints or None
+            The tuple must be of length 2, and each entry is either an int or None.
+
+            - If an int, the corresponding axis is resampled.
+            - If negative the corresponding axis is resampled in reverse
+            - If -1, the axis is inverted
+            - If 1 or None, the corresponding axis is not resampled.
+
+        transposed : bool
+            if True, the axes are swapped after resampling
+
+        Returns
+        -------
+        BivarColormap
+        """
+
+        if not np.iterable(lutshape) or len(lutshape) != 2:
+            raise ValueError("lutshape must be of length 2")
+        lutshape = [lutshape[0], lutshape[1]]
+        if lutshape[0] is None or lutshape[0] == 1:
+            lutshape[0] = self.N
+        if lutshape[1] is None or lutshape[1] == 1:
+            lutshape[1] = self.M
+
+        inverted = [False, False]
+        if lutshape[0] < 0:
+            inverted[0] = True
+            lutshape[0] = -lutshape[0]
+            if lutshape[0] == 1:
+                lutshape[0] = self.N
+        if lutshape[1] < 0:
+            inverted[1] = True
+            lutshape[1] = -lutshape[1]
+            if lutshape[1] == 1:
+                lutshape[1] = self.M
+        if not inverted[0]:
+            x_0 = np.linspace(0, 1, lutshape[0])[:, np.newaxis] * np.ones(lutshape)
+        else:
+            x_0 = np.linspace(1, 0, lutshape[0])[:, np.newaxis] * np.ones(lutshape)
+        if not inverted[1]:
+            x_1 = np.linspace(0, 1, lutshape[1])[np.newaxis, :] * np.ones(lutshape)
+        else:
+            x_1 = np.linspace(1, 0, lutshape[1])[np.newaxis, :] * np.ones(lutshape)
```

Comment:
This 2D mesh can be simplified:
```suggestion
        x_0, x_1 = np.mgrid[0:1:(lutshape[0] * 1j), 0:1:(lutshape[1] * 1j)]
        if inverted[0]:
            x_0 = x_0[::-1, :]
        if inverted[1]:
            x_1 = x_1[:, ::-1]
```